functions/send_time_off_request_to_manager/block_actions_test.ts

```diff
@@ -0,0 +1,79 @@
+import * as mf from "https://deno.land/x/mock_fetch@0.3.0/mod.ts";
+import handler from "./block_actions.ts";
+import { APPROVE_ID } from "./constants.ts";
+
+// Replaces globalThis.fetch with the mocked copy
+mf.install();
+
+mf.mock("POST@/api/chat.postMessage", async (req) => {
+  const body = await req.formData();
+  if (body.get("channel")?.toString() !== "U11111") {
+    return new Response(`{"ok": false, "error": "unexpected channel ID"}`, {
+      status: 200,
+    });
+  }
+  return new Response(`{"ok": true, "message": {"ts": "111.222"}}`, {
+    status: 200,
+  });
+});
+mf.mock("POST@/api/chat.update", () => {
+  return new Response(`{"ok": true, "message": {"ts": "111.222"}}`, {
+    status: 200,
+  });
+});
+mf.mock("POST@/api/functions.completeSuccess", () => {
+  return new Response(`{"ok": true}`, {
+    status: 200,
+  });
+});
+
+Deno.test("SendTimeOffRequestToManagerFunction runs successfully", async () => {
+  const context = {
+    action: {
+      block_id: "approve-deny-buttons",
+      action_id: APPROVE_ID,
+      style: "primary",
+      type: "button",
+      text: {
+        type: "plain_text",
+        text: "Approve",
+        emoji: true,
+      },
+      action_ts: "1664342569.823796",
+    },
+    body: {
+      container: {
+        channel_id: "C11111",
+      },
+      user: {
+        id: "U11111",
+      },
+      function_data: {
+        inputs: {
+          employee: "U11111",
+          manager: "U22222",
+          start_date: "2022-03-01",
+          end_date: "2022-03-02",
+          interactivity: {
+            interactivity_pointer: "111.222.b79....",
+            interactor: {
+              id: "U33333",
+              secret: "NDE0NTIxNDg....",
+            },
+          },
+          "interactivity.interactor": {
+            "id": "U33333",
+            "secret": "NDE0NTIxNDg....",
+          },
+          "interactivity.interactor.id": "U03E94MK0",
+          "interactivity.interactor.secret": "NDE0NTIxNDg....",
+          "interactivity.interactivity_pointer": "111.222.b79....",
+        },
+      },
+    },
+    env: { LOG_LEVEL: "ERROR" },
+    token: "valid-token",
+    // deno-lint-ignore no-explicit-any
```

Comment:
It's too hard to build a correctly typed body here... so I went with any type. Since this is a test code, it's okay to skip type checker here.

Reply:
@filmaj Yeah, perhaps we can provide something like createContext for local interactivity handlers for block_actions / view_submission / view_closed requests too.

Reply:
I spoke to the team and we will review the testing utilities and refactor them to expose a unified API for all handlers - whether function or interactivity.